### DISCUSSION TITLE
How can people propose front page redesigns? (If at all?)

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6812] [General]
@Jermolene - there has long been talk about changing the front page for tw.com . But *how* should anyone go about to make actual proposals for this?

I'm posting this question here, intentionally "hiding" it from the more public forums, because it is a topic that could easily go haywire with opinions - while in the end it is still only up to you what actually happens on tw.com

Thanx

**Comment by Jermolene** [comment]
Hi @twMat I think any major changes to the front page are going to be the result of several iterations of effort across the community. The best first step is to make a prototype/mockup, show it to the community, and get positive feedback.

**Reply by twMat** [reply]
>[...] iterations of effort across the community. [...] prototype/mockup, [...] and get positive feedback.

Thanks. But after that - then what? There's obviously no team responsible to implement it, and the twaddlers on the boards (I'm obviously one) shouting out the ideas or even making primitive mockups are typically not competent enough to implement anything of this scope.

And what about the step after *that*, i.e assume that somehow a well crafted front page has been designed in actual html/wikitext/etc - to keep it concrete, let's say it is comparable to the front of evernote.com - *then* what? Would you, as the owner of tiddlywiki.com, be willing to use it as the front page?

OR is the purpose with a community process involving "iterations, prototypes/mockups and feedback" really to give *you* ideas to implement something? Fair enough, of course.

These questions may drift a bit from the OP but the premises do affect how one, or at least I, meaningfully engage in the matter. It is desirable, perhaps even necessary, to know of constraints and preferences when making proposals.

Thanks for your input.

**Reply by Jermolene** [reply]
> Thanks. But after that - then what? There's obviously no team responsible to implement it, and the twaddlers on the boards (I'm obviously one) shouting out the ideas or even making primitive mockups are typically not competent enough to implement anything of this scope.

I think a major redesign would be squarely the responsibility of the core team. Coming up with a plan that is practical and actionable is the challenge.

> And what about the step after _that_, i.e assume that somehow a well crafted front page has been designed in actual html/wikitext/etc - to keep it concrete, let's say it is comparable to the front of evernote.com - _then_ what? Would you, as the owner of tiddlywiki.com, be willing to use it as the front page?

I don't think cloning other products is necessarily the best approach. Clearly any proposal needs my backing.

> OR is the purpose with a community process involving "iterations, prototypes/mockups and feedback" really to give _you_ ideas to implement something? Fair enough, of course.

Put differently, it's a way for us to drive consensus towards a practical way forwards.

> These questions may drift a bit from the OP but the premises do affect how one, or at least I, meaningfully engage in the matter. It is desirable, perhaps even necessary, to know of constraints and preferences when making proposals.

The only special about the front page is that it is important. The process for effecting change is the same process as for changing anything else about TiddlyWiki.

I think one problem when we discuss this stuff is actually that much of the discussion boils down to a desire to copy what we see others doing. There is a tensions here because we are fundamentally not like these other products, and wishing away the richness and complexity that makes TiddlyWiki both useful and distinctive is not the best response.

**Reply by twMat** [reply]
>I think one problem when we discuss this stuff is actually that much of the discussion boils down to a desire to copy what we see others doing. There is a tensions here because we are fundamentally not like these other products, and wishing away the richness and complexity that makes TiddlyWiki both useful and distinctive is not the best response.

So, concretely, what are your ideas in this regard? What *is* the purpose with the front page according to you? These are the constraints and preferences I refer to that would be productive to know more about. Over the years a lot of people have kept calling for, as you put it, copy what we see others doing. This is a natural to call for, because those solutions have evidently been tested and work (there). So that is what people see, and will keep suggesting.

Personally, I think we should have a front page that to the greatest extent convinces visitors, of the right kind, to try out TW properly. Very little else must distract from this, on the front page.

**Comment by kookma** [comment]
If it was me, I would use something like @saqimtiaz landing page!

Look at his landing page here: https://saqimtiaz.github.io/tw5-plugins-sandbox/

Well done Saq!

**Reply by Jermolene** [reply]
It's very good, and of course is "just" a TiddlyWiki, and so is an approach we could use for tiddlywiki.com.  The fact that it hides the regular TW UI is what makes it attractive, but of course it would also make it less useful as tiddlywiki.com.

**Reply by saqimtiaz** [reply]
If the landing page looks very different to the actual product provided, it will feel like a bait and switch and will alienate users.

My personal preference would be an interactive introduction that gradually reveals and explains the richness and complexity of TiddlyWiki and its default UI, while also showing examples of the extent of customization possible.

**Reply by kookma** [reply]
> but of course it would also make it less useful as tiddlywiki.com.

I think we should have a link button to open the documentation page and it will be a full fledge TiddlyWIki (what we have now on tiddlywiki.com).

**Reply by Jermolene** [reply]
> I think we should have a link button to open the documentation page and it will be a full fledge TiddlyWIki (what we have now on tiddlywiki.com).

That's true, but then users would need to click around to find what they are looking for. Some users might see it as hiding the docs.

The present arrangement serves the needs of several distinct sets of users:

* If you want to find out more about TW, you just need to read, scroll and click links
* If you want the docs, you can use the search box or sidebar table of contents to navigate quickly
* If you want to jump to some other TW site you can use the menu bar to jump there
* If you want to test a TW customisation, you can drag and drop a JSON file to try things out

**Reply by pmario** [reply]
I think we can make HelloThere look like other landing pages and let it be the only tiddler that is shown. I think that can be a compromise. 

I'd also like if the HelloThere tiddler would be the default tiddler, when the StoryRiver is empty. So there would be no need to click the Home button. 

I used [$:/config/EmptyStoryMessage](https://wikilabs.github.io/editions/bundler/#%24%3A%2Fconfig%2FEmptyStoryMessage) for my bundler landing-page. ... BUT If the default tiddler UI template is used, that tiddler can't be removed anymore. If the Close icon is clicked, there is no animation anymore. .. So the behaviour of the EmptyStoryMessage would need some adjustment.

**Reply by saqimtiaz** [reply]
> BUT If the default tiddler UI template is used, that tiddler can't be removed anymore. If the Close icon is clicked, there is no animation anymore.

As an aside, I vaguely recall that the story animations are dependent on the storyTiddler variable being available.

**Reply by Jermolene** [reply]
> I think we can make HelloThere look like other landing pages and let it be the only tiddler that is shown. I think that can be a compromise.

I agree that seems like a promising approach.

> I'd also like if the HelloThere tiddler would be the default tiddler, when the StoryRiver is empty. So there would be no need to click the Home button.

I think that that leads to a confusing user experience. Better to have an empty message that invites people to open tiddlers using the sidebar, and provides a copy of the "home" button.

**Reply by pmario** [reply]
>I think that that leads to a confusing user experience. Better to have an empty message that invites people to open tiddlers using the sidebar, and provides a copy of the "home" button.

That seems to be simple. .. Are we allowed to make the home icon / button big. eg: 5cm big?

**Reply by Jermolene** [reply]
> That seems to be simple. .. Are we allowed to make the home icon / button big. eg: 5cm big?

I'd be concerned that recognition would be impacted f we make the copy be too much bigger than the original. Maybe 200% or 300%.
